### DISCUSSION TITLE
System battery monitor

### DISCUSSION THREAD

**Discussion by Grover49** [feelfreelinux/octo4a#207] [Ideas]
I was wondering if ther could be a plugin made for Octo4A to monitor the battery and report charge levels and maybe even perform tasks based on parameters, pause prints, or stop the camera server.

**Comment by Zydnar** [comment]
It's not the app but the Android limitation itself. If Android has linux in permissive mode then `octo4a_battery` plugin (see: https://github.com/tobiasgraf/OctoPrint-Octo4a_battery ) can read battery status from `/sys/class/power_supply/battery/capacity`. Newest android builds mostly use linux in strict mode. Rooted android device would have access to this anyway...

**Reply by jzaor** [reply]
I tried to solve this problem by external battery monitoring service:
https://github.com/jzaor/Battery-capacity-file
